_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,241 @@
+# that is accessible by the Kubernetes cluster.
+docker build . -t <registry>/<cruise-control-ui-image-name>
+
+# Push the image to that container registry
+docker push <registry>/<cruise-control-ui-image-name>
+```
+
+### Deploying the Cruise Control UI pod
+
+Then we can create a Deployment, ConfigMap, and Service for running the Cruise Control UI:
+
+```yaml
+# cruise-control-ui.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cruise-control-ui
+  template:
+    metadata:
+      labels:
+        app: cruise-control-ui
+    spec:
+      containers:
+      - name: cruise-control-ui
+        image: <registry>/<cruise-control-ui-image-name>
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: cruise-control-ui
+          mountPath: /etc/nginx/conf.d
+      volumes:
+        - name: cruise-control-ui
+          configMap:
+            name: cruise-control-ui
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+data:
+  default.conf: |2+
+        server {
+        listen       80;
+        listen  [::]:80;
+        server_name  localhost;
+
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+
+        location /kafkacruisecontrol/ {
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_buffering off;
+            proxy_pass http://my-cluster-cruise-control:9090/kafkacruisecontrol/;
+        }
+        
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+        }
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  selector:
+    app: cruise-control-ui
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
+```
+
+Then run it:
+```
+kubectl apply -f cruise-control-ui.yaml -n <namespace>
+```
+
+## Using Cruise Control Front End
+
+After being deployed, port-forward the Cruise Control UI service to access it locally:
+
+```
+kubectl port-forward svc/cruise-control-ui 8090:80
+```
+
+Then navigate ‘http://127.0.0.1:8090’ in your browser
+
+![Cruise Control Frontend]({{ "/assets/images/posts/2019-08-08-cruise-control-frontend.png" }})
+
+## Conclusion
+
+Although it is possible to run a Cruise Control UI instance alonside a Strimzi managed Kafka cluster like this, it is still dangerous!
+To allow the Cruise Control UI access to the Cruise Control Rest API we must disable the REST API security settings.
+Having these security settings disabled gives users full access to the Cruise Control API and allows them to perform potentially destructive Cruise Control operations that are not coordinated or supported by Strimzi Operator.
```

Comment:
If the REST API is exposed. Otherwise you still need to have access to the pod/service to use it

Reply:
Are Strimzi pods/services not accessible to users by default?